### PR TITLE
System.Net.NameResolution for UWP

### PR DESCRIPTION
**Pull request by schen381**
1. Unifying Windows implementations (remove dependency on WinRT API)
2. No failing test on this namespace (including outerloop tests)

Contribute to #20139

### CODE REVIEW

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj [hunk 1]
```diff
@@ -6,9 +6,6 @@
     <ProjectGuid>{1714448C-211E-48C1-8B7E-4EE667D336A1}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
   </PropertyGroup>   
-  <PropertyGroup>
-    <DefineConstants Condition="'$(TargetGroup)' == 'uap'">$(DefineConstants);uap</DefineConstants>
```
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: What was relying on this within the code?

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj [hunk 2]
```diff
@@ -116,19 +113,10 @@
...
-  <!-- Windows : Win32 only -->
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' AND '$(TargetGroup)' != 'uap'">
     <Compile Include="$(CommonPath)\System\Net\ContextAwareResult.Windows.cs">
       <Link>Common\System\Net\ContextAwareResult.Windows.cs</Link>
     </Compile>
   </ItemGroup>
-  <!-- Windows : Win32 + WinRT -->
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' AND '$(TargetGroup)' == 'uap'">
-    <Compile Include="$(CommonPath)\System\Net\ContextAwareResult.Uap.cs">
-      <Link>Common\System\Net\ContextAwareResult.Uap.cs</Link>
```
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: We'll need to remove this source file after we have no other references. 
What else is referencing it?

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj [hunk 3]
```diff
@@ -116,19 +113,10 @@
     <Compile Include="$(CommonPath)\Interop\Windows\Winsock\SafeFreeAddrInfo.cs">
       <Link>Interop\Windows\Winsock\SafeFreeAddrInfo.cs</Link>
     </Compile>
-  </ItemGroup>
-  <!-- Windows : Win32 only -->
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' AND '$(TargetGroup)' != 'uap'">
     <Compile Include="$(CommonPath)\System\Net\ContextAwareResult.Windows.cs">
```
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: Nit: I think this file would be better placed somewhere in line 52 to match the comments.

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 2 above)
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: We have a few similar items. Let's track all with #20212.

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 1 above)
Comment (schen381) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: There is nothing in the code rely on it (no #if uap statement), and there is no reference to UWP.

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 3 above)
Comment (schen381) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: sure, will fix.

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 2 above)
Comment (schen381) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: Apart from S.N.NameResolution, System.Net.Sockets is the only one referencing it.

Comment (schen381) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: Should I remove the file in this PR?

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 1 above)
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: >  nothing in the code rely on it

The file is in Common: have you checked all other components? (You can try deleting it and do a full build to see if anything else depends on it.)

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 2 above)
Comment (CIPop) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: How would Sockets build in that case?

Comment (schen381) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: I will look into it.

**Comment by schen381** [comment]
The new commit should fix #20139. I will investigate more on ContextAwareResult.Uap.cs and post status at #20212.

**Comment by ericstj** [comment]
You added a UAP configuration to the project files but didn't actually update Configurations.props.  This will get removed the next time some runs build /t:UpdateVSConfigurations.  If you want the tests to cross-compile for UAP you should update Configurations.props.  See https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md#aggregate-build-properties

**Comment by ericstj** [comment]
I actually don't see you doing anything UAP specific, so you can keep all the existing netstandard builds for the tests and just run them for UAP.  The same document describes how to run for UAP.

**Pull request merged**